### PR TITLE
Add Access Methods for Authenticator API Endpoints

### PR DESCRIPTION
**Pull request by micahlee**
Connected to #157 

Add Ruby API methods for accessing the Conjur API endpoints for authenticator status and whitelist management.

### CODE REVIEW

File: features/authenticators.feature [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    And the JSON should have "enabled"
+    And the JSON at "enabled" should have 1 entry
+
+  Scenario: Enable and disable authenticator
+    When I run the code:
+    """
+    $conjur.authenticator_enable("authn-k8s", "my-auth")
+    $conjur.authenticator_list
+    """
+    Then the JSON at "enabled" should have 2 entries
```
Comment (jtuttle) [inline features/authenticators.feature]: Can we be more explicit about what those entries actually are?

File: lib/conjur/api/authenticators.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+require 'conjur/webservice'
+
+module Conjur
+  # API contains each of the methods for access the Conjur API endpoints
+  #-- :reek:DataClump for authenticator identifier fields (name, id, account)
+  class API
+    # @!group Authenticators
+
+    # List all configured databases
```
Comment (jtuttle) [inline lib/conjur/api/authenticators.rb]: Typo: "databases" => "authenticators"

File: lib/conjur/api/authenticators.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    def authenticator_list
+      JSON.parse(url_for(:authenticators).get)
+    end
+
+    # Enables an authenticator in Conjur. The authenticator must be defined and
+    # loading in Conjur policy prior to enabling it.
```
Comment (jtuttle) [inline lib/conjur/api/authenticators.rb]: Typo: "loading" => "loaded"

File: features/authenticators.feature (hunk 1 above)
Comment (micahlee) [reply features/authenticators.feature]: We can! I found a better test for this in the JSON cuke helpers.

File: features/authenticators.feature [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    $conjur.authenticator_enable("authn-k8s", "my-auth")
+    $conjur.authenticator_list
+    """
+    Then the JSON at "enabled" should be ["authn", "authn-k8s/my-auth"]
+    When I run the code:
+    """
+    $conjur.authenticator_disable("authn-k8s", "my-auth")
+    $conjur.authenticator_list
+    """
+    Then the JSON at "enabled" should have 1 entry
```
Comment (jtuttle) [inline features/authenticators.feature]: Can we make this one explicit as well?

File: features/authenticators.feature [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+  Scenario: Authenticator list includes the authenticator status
+    When I run the code:
+    """
+    $conjur.authenticator_list
+    """
+    Then the JSON should have "installed"
+    And the JSON should have "configured"
+    And the JSON should have "enabled"
+    And the JSON at "enabled" should have 1 entry
```
Comment (jtuttle) [inline features/authenticators.feature]: Can we make this one explicit as well?

**Pull request merged**